src/Illuminate/Testing/TestResponse.php

```diff
@@ -46,7 +46,7 @@ class TestResponse implements ArrayAccess
     /**
      * The response to delegate to.
      *
-     * @var \Illuminate\Http\Response
+     * @var \Symfony\Component\HttpFoundation\Response
```

Comment:
I believe it should be
```suggestion
     * @var \Illuminate\Http\Response|\Symfony\Component\HttpFoundation\StreamedJsonResponse|\Symfony\Component\HttpFoundation\StreamedResponse
```

https://github.com/laravel/framework/blob/b17b2bd8ebcb850a909fd046c380d075fd2f118b/src/Illuminate/Testing/TestResponse.php#L1623-L1626

Reply:
@crynobone When the response is a redirection, the response type is wrong.
e.g.
```php
TestResponse::fromBaseResponse(response()->redirectTo('')
The response type is `\Illuminate\Http\RedirectResponse`
```

Reply:
@crynobone  Since all the different response classes are based on `Symfony\Component\HttpFoundation\Response`, it is more practical to use the base class in the PHPDoc.

Generalization: Using the base class covers all possible response types, making the code more flexible.

Simplicity: It simplifies the type declarations by avoiding the need to list all possible response classes.

Consistency: It ensures that static analysis tools can correctly interpret the type, providing accurate type hints and avoiding potential omissions.

By using `Symfony\Component\HttpFoundation\Response`, we maintain a cleaner and more maintainable codebase.

Reply:
The reason I suggested above is that `TestResponse` also provides magic methods including `__call()` which currently expects methods from `Illuminate\Http\Response` to be available. While this would fix your issue it would then cause errors when users depend on those methods to be available and using PHPStan etc